### PR TITLE
Remove lib/QueueAsync

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

I wanted to convert another module into ES6, but it turned out, it would be easier to remove it at all.

## What is the value of this and can you measure success?

Less code & less cognitive load.

## Does this affect other platforms - Amp, Apps, etc?

No.

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/article/space-filler.js [hunk 1]
```diff
@@ -32,7 +28,7 @@ define([
      */
     SpaceFiller.prototype.fillSpace = function (rules, writer, options) {
         var write = (options && options.domWriter) || fastdom.write;
-        return queue.add(insertNextContent);
+        return this.queue.then(insertNextContent).catch(onError);
```
Comment (regiskuckaertz) [inline static/src/javascripts-legacy/projects/common/modules/article/space-filler.js]: You will also have to assign the result to `this.queue`

**Review by regiskuckaertz — CHANGES_REQUESTED** [review]
`then` and `catch` return new promises, so you need to keep track of that.

File: static/src/javascripts-legacy/projects/common/modules/article/space-filler.js (hunk 1 above)
Comment (gustavpursche) [inline static/src/javascripts-legacy/projects/common/modules/article/space-filler.js]: You are right - I didn't think about that the promises should run in the right order. 👍

**Review by sndrs — APPROVED** [review]

**Review by regiskuckaertz — APPROVED** [review]

**Comment by regiskuckaertz** [comment]
Ace 👍

**Pull request merged**